client/landing/stepper/declarative-flow/hosted-site-migration-v2/index.tsx

```diff
@@ -11,14 +18,75 @@ export default {
 	},
 	isSignupFlow: false,
 	useSteps() {
-		return stepsWithRequiredLogin( [ STEPS.IMPORT_OR_MIGRATE_WITH_UPGRADE ] );
+		return stepsWithRequiredLogin( [
+			PLATFORM_IDENTIFICATION,
+			SITE_CREATION_STEP,
+			PROCESSING,
+			SITE_MIGRATION_UPGRADE_PLAN,
+		] );
 	},
 
-	//TEMP: This is a temporary implementation to make the flow work
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
 	useStepNavigation( currentStep, navigate ) {
+		const [ query ] = useSearchParams();
+
 		return {
-			submit: () => alert( 'Go' ),
+			submit: ( props?: ProvidedDependencies ) => {
+				const data = {
+					platform: props?.platform || query.get( 'platform' ),
+					siteId: props?.siteId || query.get( 'siteId' ),
+					siteSlug: props?.siteSlug || query.get( 'siteSlug' ),
+					next: props?.next || query.get( 'next' ),
+					url: props?.url,
+				} as Record< string, Primitive >;
+
+				if ( currentStep === PLATFORM_IDENTIFICATION.slug ) {
+					return navigate(
+						addQueryArgs(
+							{
+								platform: data.platform,
+								next: data.url,
+							},
+							STEPS.SITE_CREATION_STEP.slug
+						)
+					);
+				}
+
+				if ( currentStep === SITE_CREATION_STEP.slug ) {
+					return navigate(
+						addQueryArgs(
+							{
+								platform: data.platform,
+								next: data.next,
+							},
+							STEPS.PROCESSING.slug
+						)
+					);
+				}
+
+				if ( currentStep === PROCESSING.slug ) {
+					if ( data?.next ) {
+						return window.location.assign(
```

Comment:
I see that the `hosted-site-migration` also does the same, redirecting the flow to the `/setup/site-setup`. Just for my understanding, is there a special reason we do this instead of adding all the steps in this flow?

Reply:
Although the steps are very reusable the logic to continue on each user interaction (site-setup.ts)   is not easy to reuse. In these cases we are redirecting the users to the more appropriate flow.